Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -265,93 +276,123 @@ namespace AZ::SceneAPI::Behaviors
             return result;
         }
 
-        NodeEntityMap CreateMeshGroups(
+        bool CreateMeshGroupAndComponents(
             ManifestUpdates& manifestUpdates,
-            const MeshDataMap& meshDataMap,
+            AZ::EntityId entityId,
+            const NodeDataForEntity& nodeData,
+            const NodeDataMap& nodeDataMap,
             const Containers::Scene& scene,
             const AZStd::string& relativeSourcePath)
         {
-            NodeEntityMap nodeEntityMap;
-            const auto& graph = scene.GetGraph();
+            const auto meshNodeIndex = nodeData.m_meshIndex;
+            const auto propertyDataIndex = nodeData.m_propertyMapIndex;
 
-            for (const auto& entry : meshDataMap)
+            const auto& graph = scene.GetGraph();
+            const auto meshNodeName = graph.GetNodeName(meshNodeIndex);
+            const auto meshSubId =
+                DataTypes::Utilities::CreateStableUuid(scene, azrtti_typeid<AZ::SceneAPI::SceneData::MeshGroup>(), meshNodeName.GetPath());
+
+            AZStd::string meshGroupName = "default_";
+            meshGroupName += scene.GetName();
+            meshGroupName += meshSubId.ToFixedString().c_str();
+
+            // clean up the mesh group name
+            AZStd::replace_if(
+                meshGroupName.begin(),
+                meshGroupName.end(),
+                [](char c)
+                {
+                    return (!AZStd::is_alnum(c) && c != '_');
+                },
+                '_');
+
+            AZStd::string meshNodePath{ meshNodeName.GetPath() };
+            auto meshGroup = AZStd::make_shared<AZ::SceneAPI::SceneData::MeshGroup>();
+            meshGroup->SetName(meshGroupName);
+            meshGroup->GetSceneNodeSelectionList().AddSelectedNode(AZStd::move(meshNodePath));
```

Comment:
This combines all the meshes into a single mesh group. 
This is directly against the idea of breaking apart the mesh data nodes into their own mesh groups.
Think of a car scene with separate wheels and a single body to be re-constructed in the default procedural prefab. 
Please revert this logic.